docs/openstack/heat_adoption.md

```diff
@@ -0,0 +1,158 @@
+# Heat adoption
+
+Adopting Heat means that an existing `OpenStackControlPlane` CR, where Heat
+is supposed to be disabled, should be patched to start the service with the
+configuration parameters provided by the source environment.
+
+After the adoption process has been completed, a user can expect that they
+will then have CR's for `Heat`, `HeatAPI`, `HeatEngine` and `HeatCFNAPI`.
+Additionally, a user should have endpoints created within Keystone to facilitate
+the above mentioned servies.
+
+This guide also assumes that:
+
+1. A `TripleO` environment (the source Cloud) is running on one side;
+2. A OpenShift environment is running on the other side.
+
+## Prerequisites
+
+* Previous Adoption steps completed. Notably, MariaDB and Keystone
+  should be already adopted.
+* In addition, if your existing Heat stacks contain resources from other services
+  such as Neutron, Nova, Swift, etc. Those services should be adopted first before
+  trying to adopt Heat.
+
+## Procedure - Heat adoption
+
+As already done for [Keystone](https://github.com/openstack-k8s-operators/data-plane-adoption/blob/main/keystone_adoption.md), the Heat Adoption follows a similar pattern.
+
+Patch the `osp-secret` to update the `HeatAuthEncryptionKey` password. This needs
+to match what you have configured in the existing TripleO Heat configuration.
+
+```
+Insert Example of how to get this from the overcloud passwords.
+```
+
+Patch OpenStackControlPlane to deploy Heat:
+
+```
+oc patch openstackcontrolplane openstack --type=merge --patch '
```

Comment:
And this could have a corresponding new Ansible role `heat_adoption` in the test suite:

https://github.com/openstack-k8s-operators/data-plane-adoption/tree/main/tests/roles

included from both playbooks:

https://github.com/openstack-k8s-operators/data-plane-adoption/tree/main/tests/playbooks

The Standalone should now be deploying Heat, so it should be adoptable in CI i think.

https://github.com/openstack-k8s-operators/install_yamls/blob/c5e089ca59b3c15e28e9b65d968b238f573cfe00/devsetup/standalone/role.j2#L98-L101

Reply:
Let me know if the latest revision to this PR addresses these now.

Reply:
Thanks, overall it looks good to me. I think before we fork into TP3 docs there may be opportunities to streamline the text a bit but i'd prefer getting something working merged over polishing details. (E.g. I'd just trust that the user set the `osp-secret` passwords correctly as the relevant document says. It's also a matter of consistency -- we don't tell the user to double check those secrets in any of the other service adoption documents.)